http.go

```diff
@@ -56,6 +61,12 @@ func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.R
 
 	infoLog.Log(messageKey, "Receiving incoming request...")
 
+	// find time difference, add to metric after function finishes
+	defer func() {
```

Comment:
`defer` closures that are inline can get a bit hard to read after a while, especially as they are added over time.

A cleaner way is just pull out the `defer` into an unexported method.  This has the extra side effect of making the defer logic separately testable.

For example:

```
// this function can now have separate unit tests, if applicable
// especially as the codebase changes over time, this separate function is easier to deal with
func (sh *ServerHandler) recordQueueLatency(startTime time.Time) {
    endTime := sh.now()
    sh.incomingQueueLatency.With("event", eventType).Observe(endTime.Sub(startTime).Seconds())
}

func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
    startTime := sh.now()
    defer sh.recordQueueLatency(startTime)

    // if you don't need the startTime variable below, you can just abbreviate the above to:
    defer sh.recordQueueLatency(sh.now())

    // the rest of the code as is
}
```